types/rational_test.go

```diff
@@ -315,3 +316,81 @@ func TestStringOverflow(t *testing.T) {
 		rat3.String(),
 	)
 }
+
+func TestArithRat(t *testing.T) {
+	for i := 0; i < 20; i++ {
+		n1 := NewInt(int64(rand.Int31()))
```

Comment:
Previous comments about randomized testing seed logging apply to this file as well. 

Could you repeat the same note you made about why we can use random testing for ints here?